src/multitask_personalization/envs/cooking/cooking_hidden_spec.py

```diff
@@ -80,6 +85,71 @@ def __init__(self, meal_specs: list[MealSpec]) -> None:
                     quant_lo,
                     quant_hi,
                 )
+        # Lifelong learning setup.
+        # If a shift occurs, all ingredients will shift at the same time.
+        # The amount is sampled within a range.
+        # number of feedbacks given as a proxy for number of meals user had
+        # since the last shift
+        self._n_feedbacks_given = 0
+
+        self._min_shift_interval = (
+            5  # minimum number of meals before a potential preference shift
+        )
+        self._shift_prob = 0.2  # probability of a preference shift
+        # range of the shift factor.
+        # Given old range [x-r, x+r], new range is [max(0, x*f-r), x*f+r]
+        self._shift_factor_range = (
+            0.0,
+            2.0,
+        )
+
+    def shift_preferences(self, rng: np.random.Generator) -> None:
```

Comment:
this looks good but can you add a small unit test? :) it doesn't need to test anything thoroughly -- basically I just want to make sure that this new code is run and doesn't crash (and continues to run and not crash for any future changes)